### ISSUE TITLE
Warning when compile greenplum master on mac Mojave

### ISSUE THREAD

**Issue by yydzero** [greenplum-db/gpdb#7314]
### Step to reproduce the behavior

ginpostinglist.c:89:24: warning: result of comparison of constant 65536 with expression of type 'OffsetNumber' (aka 'unsigned short') is always true
      [-Wtautological-constant-out-of-range-compare]
        Assert(iptr->ip_posid < (1 << MaxHeapTuplesPerPageBits));
        ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/include/c.h:784:10: note: expanded from macro 'Assert'
                Trap(!(condition), "FailedAssertion")
                ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/include/c.h:766:28: note: expanded from macro 'Trap'
                if ((assert_enabled) && (condition)) \
                                         ^~~~~~~~~
1 warning generated.

**Comment by danielgustafsson** [comment]
This is due to the expanded offsetnumber for GIN postinglists in Greenplum which landed recently, fixed in the linked PR.

**Comment by hlinnaka** [comment]
Closing, since this was fixed by PR #7319.
